src/graphnet/models/graphs/graph_definition.py

```diff
@@ -166,7 +166,7 @@ def _validate_input(
         for idx in range(len(node_feature_names)):
             assert (
                 node_feature_names[idx] == self._node_feature_names[idx]
-            ), """ Order of node features are not the same."""
+            ), f""" Order of node features are not the same. Got {node_feature_names} vs. {self._node_feature_names}"""
```

Comment:
I like including more information in this assert statement. However the additional context doesn't immediately clarify to me which name is from the config file and which is the native name of the detector.

I also think your suggested wording is misleading, since `node_feature_names` are usually the names from the detector, whereas `self._node_feature_names` are usually defined from a config file.

Reply:
This assert statement tries to catch cases where the `features` argument to the `Dataset` is of a different ordering than the `features` argument given to the `GraphDefinition`. Maybe we should consider removing the `features` argument for the `Dataset` class and simply let it extract it from the `GraphDefinition`. Anyway - I've updated error message to be a bit clearer.